### ISSUE TITLE
[BUG] UnpicklingError while fitting model

### ISSUE THREAD

**Issue by jbottke** [basf/mamba-tabular#212]
**Describe the bug**
When trying to fit a Mambular classification model on the adult census dataset, the training process results in an UnpicklingError.

**To Reproduce**
- load adult census dataset using shapiq with shapiq.datasets.load_adult_census()
- preprocess data using sklearns train_test_split() function which shuffles dataset und splits it into four numpy arrays (x_train, x_test, y_train, y_test)
- create Instance of MambularClassifier and fit it on numpy arrays x_train, y_train 

**Expected behavior**
Method is expected to train a Mambular classification model and save the best model under model_checkpoints as a .ckpt file. 

**Screenshots**

<img width="778" alt="Image" src="https://github.com/user-attachments/assets/92df4994-9519-4e50-af1d-50f18c8589e7" />

**Desktop:**
 - OS: Ubuntu via WSL2 
 - Python version 3.10.12
 - Mambular Version 1.1.0

**Comment by AnFreTh** [comment]
Thanks for raising this issue! Could you please add your PyTorch and lightning versions?

**Comment by jbottke** [comment]
- lightning 2.5.0.post0
- lightning-utilities 0.12.0
- pytorch-lightning 2.5.0.post0
- torch 2.6.0
- torchmetrics 1.6.1

**Comment by AnFreTh** [comment]
@mkumar73 Restricting torch to 2.2.2 - 2.5.1 should solve the issue.

**Comment by AnFreTh** [comment]
Thanks again for raising this issue! The latest release v1.2.0 #222 should solve this issue. Feel free to reopen or open any other issues if you encounter any more bugs.
